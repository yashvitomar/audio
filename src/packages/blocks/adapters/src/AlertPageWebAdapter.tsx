import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class AlertPageWebAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    console.log('initing AlertPageWebAdapter');
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationAlertWebMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(getName(MessageEnum.NavigationTargetMessage), 'AlertWeb');

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    const raiseMessage: Message = new Message(
      getName(MessageEnum.NavigationPayLoadMessage)
    );

    raiseMessage.copyAllPropertiesOf(from);

    to.addData(getName(MessageEnum.NavigationRaiseMessage), raiseMessage);

    return to;
  };

  receive(from: string, message: Message): void {
    console.log(`receive AlertPageWebAdapter==>${JSON.stringify(message)}`);
    console.log(
      `Converted AlertPageWebAdapter==>${JSON.stringify(this.convert(message))}`
    );

    this.send(this.convert(message));
  }
}
