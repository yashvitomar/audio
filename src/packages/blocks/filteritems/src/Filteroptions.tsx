import React from 'react';
// Customizable Area Start
import {
  StyleSheet,
  Text,
  TouchableOpacity,
  View,
  SafeAreaView,
  Dimensions,
  TextInput,
  FlatList,
  Platform,
} from 'react-native';
const windowWidth = Dimensions.get('window').width;
const windowHeight = Dimensions.get('window').height;
import Icon from 'react-native-vector-icons/FontAwesome';
import MultiSlider from '@ptomasroos/react-native-multi-slider';
import CheckBox from '@react-native-community/checkbox';
// Customizable Area End

import FilteroptionsController, {
  Props,
  //configJSON
} from './FilteroptionsController';

export default class Filteroptions extends FilteroptionsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  getBrands(item: any, index: any) {
    let value = item.attributes;
    return (
      <View style={[styles.flexBox, styles.checkBox]} key={item.id}>
        <CheckBox
          key={value.id}
          value={item.checked}
          boxType={'square'}
          style={Platform.OS === 'ios' ? styles.innerCheckBox : null}
          animationDuration={0}
          onValueChange={() => this.onCheck(item, index)}
          testID={'getBrands'}
        />
        <Text style={styles.checkText}> {value.name}</Text>
      </View>
    );
  }
  getCategory(item: any, index: any) {
    let value = item.attributes.category;
    return (
      <View style={[styles.flexBox, styles.checkBox]} key={item.id}>
        <CheckBox
          key={value.id}
          value={item.checked}
          boxType={'square'}
          style={Platform.OS === 'ios' ? styles.innerCheckBox : null}
          animationDuration={0}
          onValueChange={() => this.onCheckCategory(item, index)}
          testID={'getCategory'}
        />
        <Text style={styles.checkText}> {value.name}</Text>
      </View>
    );
  }
  onCheck = (item: any, i: any) => {
    let items = [];
    items = this.state.GetAllBrand;
    items[i].checked = items[i].checked ? !items[i].checked : true;
    console.log(items, 'item.checked');
    if (items) {
      items.forEach((d: any) => {
        if (item.id === d.id) {
          if (d.checked) {
            this.state.selectedItems.push(d);
          } else {
            this.setState({ selectedItems: [] });
          }
        }
      });
    }

    this.setState({ GetAllBrand: items });
  };
  onCheckCategory = (item: any, i: any) => {
    let items = [];
    items = this.state.arrayHolder;
    items[i].checked = items[i].checked ? !items[i].checked : true;
    console.log(items, 'item.checked');
    if (items) {
      items.forEach((d: any) => {
        if (item.id === d.id) {
          console.log(item.id, 'item.id', d.id, 'd.id');
          if (d.checked) {
            this.state.selectedCategory.push(d);
          } else {
            this.setState({
              selectedCategory: [],
            });
          }
        }
      });
    }

    this.setState({ data: items });
  };
  searchFilterFunction = (text: any) => {
    if (this.state.brand) {
      if (text === '') {
        this.setState({
          GetAllBrand: this.state.BrandList,
        });
      } else {
        let Name = this.state.BrandList;
        const textData = Name.filter((item: any) =>
          item.attributes.name
            .toLocaleLowerCase()
            .startsWith(text.toLowerCase())
        );
        this.setState({ GetAllBrand: textData });
      }
    } else if (this.state.category) {
      if (text === '') {
        this.setState({
          arrayHolder: this.state.categoryArray,
        });
      } else {
        let Name = this.state.categoryArray;
        const textData = Name.filter((item: any) =>
          item.attributes.category.name
            .toLocaleLowerCase()
            .startsWith(text.toLowerCase())
        );
        this.setState({ arrayHolder: textData });
      }
    }
  };

  valueChange(value: any) {
    this.setState({
      priceMin: value[0],
      priceMax: value[1],
    });
  }

  valueChangeFinish() {
    this.setState({
      minValue: this.state.priceMin,
      maxValue: this.state.priceMax,
    });
  }

  renderPriceRange() {
    if (!this.state.pricerange) {
      return null;
    }
    const { minValue, maxValue } = this.state;
    const sliderLength = windowWidth - 200;
    const min =
      this.props.navigation.state.params &&
      this.props.navigation.state.params.min
        ? this.props.navigation.state.params.min
        : 0;
    const max =
      this.props.navigation.state.params &&
      this.props.navigation.state.params.max
        ? this.props.navigation.state.params.max
        : 100;
    return (
      <View style={styles.sliderContainer}>
        <MultiSlider
          onValuesChange={(value) => this.valueChange(value)}
          onValuesChangeFinish={() => this.valueChangeFinish()}
          sliderLength={sliderLength}
          values={[minValue, maxValue]}
          min={min}
          max={max}
          enableLabel={true}
          step={1}
        />
      </View>
    );
  }
  renderBrandList() {
    const { GetAllBrand } = this.state;
    if (!this.state.brand) {
      return null;
    }
    return (
      <View>
        <View style={[styles.flexBox, styles.textBox]}>
          <Icon
            name='search'
            size={15}
            style={styles.searchIcon}
            color='#808080'
          />
          <TextInput
            style={styles.inputStyle}
            placeholder='Search for a Brand'
            onChangeText={(text) => this.searchFilterFunction(text)}
            testID={'filterBrand'}
          />
        </View>
        <FlatList
          renderItem={({ item, index }) => this.getBrands(item, index)}
          contentContainerStyle={styles.flatContainer}
          scrollEnabled={true}
          extraData={this.state}
          keyExtractor={(item: any) => item.id}
          data={GetAllBrand}
        />
      </View>
    );
  }
  renderCategoryList() {
    if (!this.state.category) {
      return null;
    }
    return (
      <View>
        <View style={[styles.flexBox, styles.textBox]}>
          <Icon
            name='search'
            size={15}
            style={styles.searchIcon}
            color='#808080'
          />
          <TextInput
            style={styles.inputStyle}
            placeholder='Search for a Category'
            onChangeText={(text) => this.searchFilterFunction(text)}
            testID={'filterCategory'}
          />
        </View>
        <FlatList
          renderItem={({ item, index }) => this.getCategory(item, index)}
          contentContainerStyle={styles.flatContainer}
          scrollEnabled={true}
          extraData={this.state}
          keyExtractor={(item: any) => item.id}
          data={this.state.arrayHolder}
        />
      </View>
    );
  }

  updateItemsCheckedStatus = () => {
    const { GetAllBrand, selectedItems, data, selectedCategory } = this.state;

    if (selectedItems && selectedItems.length > 0 && GetAllBrand) {
      const updatedBrandList = GetAllBrand.map((item: any) => ({
        ...item,
        checked: !!selectedItems.find((o: any) => o.id === item.id),
      }));
      this.setState({
        GetAllBrand: updatedBrandList,
      });
    }

    if (selectedCategory && selectedCategory.length > 0 && data) {
      const updatedCategoryList = data.map((item: any) => ({
        ...item,
        checked: !!selectedCategory.find((o: any) => o.id === item.id),
      }));
      this.setState({
        data: updatedCategoryList,
      });
    }
  };
  //Customizable Area End

  render() {
    // Customizable Area Start
    // Merge Engine - render - Start
    this.updateItemsCheckedStatus();

    return (
      <SafeAreaView style={styles.container}>
        <View>
          <View style={[styles.flexBox, styles.containerBox]}>
            <View style={styles.leftContent}>
              <TouchableOpacity
                style={[
                  styles.textBtn,
                  this.state.pricerange ? styles.bgWhite : null,
                ]}
                onPress={() => this.openFilter('pricerange')}
                testID={'openFilterPrice'}
              >
                <Text style={styles.textButton}>Price range</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={[
                  styles.textBtn,
                  this.state.brand ? styles.bgWhite : null,
                ]}
                onPress={() => this.openFilter('brand')}
                testID={'openFilterBrand'}
              >
                <Text style={styles.textButton}>Brand</Text>
              </TouchableOpacity>
              <TouchableOpacity
                style={[
                  styles.textBtn,
                  this.state.category ? styles.bgWhite : null,
                ]}
                onPress={() => this.openFilter('category')}
                testID={'openFiltercategory'}
              >
                <Text style={styles.textButton}>Category</Text>
              </TouchableOpacity>
            </View>
            <View style={styles.rightContent}>
              {this.renderPriceRange()}
              {this.renderBrandList()}
              {this.renderCategoryList()}
            </View>
          </View>
          <View style={[styles.flexBox, styles.buttonBox]}>
            <TouchableOpacity
              style={styles.cancelbtn}
              onPress={() => this.props.navigation.goBack()}
              testID={'BackPage'}
            >
              <Text style={styles.btnText}>Cancel</Text>
            </TouchableOpacity>
            <TouchableOpacity
              style={[styles.applybtn]}
              onPress={() => this.applyAllfilters()}
              testID={'applyFilter'}
            >
              <Text style={styles.btnText}>Apply</Text>
            </TouchableOpacity>
          </View>
        </View>
      </SafeAreaView>
    );
    // Merge Engine - render - End
    // Customizable Area End
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginLeft: 'auto',
    marginRight: 'auto',
    width: '100%',
    maxWidth: 650,
    backgroundColor: '#ffffffff',
  },
  row: {
    flex: 1,
    justifyContent: 'space-between',
  },
  flexBox: {
    display: 'flex',
    flexDirection: 'row',
  },
  leftContent: {
    width: '35%',
    backgroundColor: '#ddd',
    height: '100%',
    paddingVertical: 15,
    borderRightWidth: 1,
    borderRightColor: '#ccc',
  },
  rightContent: {
    width: '65%',
    height: '100%',
    paddingVertical: 15,
    paddingHorizontal: 15,
  },
  textBtn: {
    paddingVertical: 15,
    paddingHorizontal: 10,
  },
  textButton: {
    fontSize: 16,
    color: '#444',
  },
  containerBox: {
    height: windowHeight - 150,
  },
  checkText: {
    fontSize: 16,
  },
  checkBox: {
    alignItems: 'center',
    marginVertical: 5,
  },
  innerCheckBox: {
    height: 25,
    width: 25,
  },
  bgWhite: {
    backgroundColor: '#fff',
  },
  cancelbtn: {
    backgroundColor: '#ccc',
    width: '49%',
    borderRadius: 6,
    padding: 10,
  },
  btnText: {
    color: 'white',
    fontWeight: 'bold',
    fontSize: 17,
    textAlign: 'center',
  },
  applybtn: {
    backgroundColor: 'green',
    width: '49%',
    borderRadius: 6,
    padding: 10,
  },
  flatContainer: {
    paddingBottom: 50,
  },
  buttonBox: {
    padding: 15,
    justifyContent: 'space-between',
    borderTopWidth: 1,
    borderTopColor: '#ccc',
    backgroundColor: '#FFF',
  },
  searchIcon: {},
  inputStyle: {
    width: '100%',
    paddingLeft: 25,
  },
  textBox: {
    width: '100%',
    position: 'relative',
    alignItems: 'center',
    marginBottom: 10,
    borderBottomColor: '#ccc',
    borderBottomWidth: 1,
    paddingBottom: 10,
  },
  sliderContainer: {
    width: '100%',
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'center',
    marginTop: 60,
  },

  // disableButton:{
  //   opacity:0.5
  // },
  // activeButton:{
  //   opacity:1
  // }
});
// Customizable Area End
