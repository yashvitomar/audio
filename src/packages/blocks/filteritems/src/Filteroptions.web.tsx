import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  Platform
} from "react-native";
// Customizable Area End

import FilteroptionsController, {
  Props,
} from "./FilteroptionsController";

export default class Filteroptions extends FilteroptionsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  // Customizable Area End

  render() {
    return (
      //Merge Engine DefaultContainer
      <ScrollView keyboardShouldPersistTaps="always" style={styles.container}>
        <TouchableWithoutFeedback
          onPress={() => {
            this.hideKeyboard();
          }}
        >
          {/* Customizable Area Start */}
          {/* Merge Engine UI Engine Code */}
          <View>
            {this.isPlatformWeb() ? (
              <Text
                testID="labelTitle" //Merge Engine::From BDS
                style={styles.title} //UI Engine::From Sketch
              >
                Web Filter Options
              </Text> //UI Engine::From Sketch
            ) : null}
            <Text
              testID="labelBody" //Merge Engine::From BDS
              style={styles.body} //UI Engine::From Sketch
            >
              {" "}
              {/* UI Engine::From Sketch */}
              Empty Block
            </Text>
          </View>
          {/* Merge Engine UI Engine Code */}
          {/* Customizable Area End */}
        </TouchableWithoutFeedback>
      </ScrollView>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16,
    marginLeft: "auto",
    marginRight: "auto",
    width: Platform.OS === "web" ? "75%" : "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff"
  },
  title: {
    marginBottom: 32,
    fontSize: 16,
    textAlign: "left",
    marginVertical: 8
  },
  body: {
    marginBottom: 32,
    fontSize: 16,
    textAlign: "left",
    marginVertical: 8
  },
  bgPasswordContainer: {
    flexDirection: "row",
    backgroundColor: "#00000000",
    marginBottom: 16,
    borderBottomWidth: 1,
    borderColor: "#767676",
    borderRadius: 2,
    padding: 10,
    borderWidth: Platform.OS === "web" ? 0 : 1
  },
  bgMobileInput: {
    flex: 1
  },
  showHide: {
    alignSelf: "center"
  },
  imgShowhide: Platform.OS === "web" ? { height: 30, width: 30 } : {}
});
// Customizable Area End
