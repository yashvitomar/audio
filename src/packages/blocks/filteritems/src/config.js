Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.apiMethodTypeGet = "GET";
exports.productAPiEndPoint = "catalogue/catalogues";
exports.brandAPiEndPoint = "catalogue/brands";
exports.filterbrandAPiEndPoint = "filter_items/filtering?q[brand_id]=";
exports.singleCategoryAPiEndPoint = "filter_items/filtering?q[category_id]=";
exports.cataloguesAPiEndPoint = "filter_items/filtering?";
// Customizable Area End