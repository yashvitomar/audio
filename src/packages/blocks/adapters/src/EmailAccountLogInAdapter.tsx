import { v4 as uuidv4 } from 'uuid';
import { runEngine } from '../../../framework/src/RunEngine';
import { IBlock } from '../../../framework/src/IBlock';
import { Message } from '../../../framework/src/Message';
import MessageEnum, {
  getName
} from '../../../framework/src/Messages/MessageEnum';

export default class EmailAccountLogInAdapter {
  send: (message: Message) => void;

  constructor() {
    const blockId = uuidv4();
    this.send = message => runEngine.sendMessage(blockId, message);
    runEngine.attachBuildingBlock(this as IBlock, [
      getName(MessageEnum.NavigationEmailLogInMessage)
    ]);
  }

  convert = (from: Message): Message => {
    const to = new Message(getName(MessageEnum.NavigationMessage));

    to.addData(
      getName(MessageEnum.NavigationTargetMessage),
      'EmailAccountLoginBlock'
    );

    to.addData(
      getName(MessageEnum.NavigationPropsMessage),
      from.getData(getName(MessageEnum.NavigationPropsMessage))
    );

    return to;
  };

  receive(from: string, message: Message): void {
    this.send(this.convert(message));
  }
}
