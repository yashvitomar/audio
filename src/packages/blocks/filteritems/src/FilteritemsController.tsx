import { IBlock } from "../../../framework/src/IBlock";
import { Message } from "../../../framework/src/Message";
import { BlockComponent } from "../../../framework/src/BlockComponent";
import MessageEnum, {
  getName
} from "../../../framework/src/Messages/MessageEnum";
import { runEngine } from "../../../framework/src/RunEngine";

// Customizable Area Start
// Customizable Area End

export const configJSON = require("./config");

export interface Props {
  navigation: any;
  id: string;
  // Customizable Area Start
  // Customizable Area End
}

interface S {
  // Customizable Area Start
  token: string;
  data: any;
  arrayHolder: any
  // Customizable Area End
}

interface SS {
  id: any;
}

export default class FilteritemsController extends BlockComponent<
  Props,
  S,
  SS
> {
  getProductApiCallId: any;
  constructor(props: Props) {
    super(props);
    this.receive = this.receive.bind(this);

    // Customizable Area Start
    this.subScribedMessages = [
      getName(MessageEnum.AccoutLoginSuccess),
      getName(MessageEnum.RestAPIResponceMessage),
      getName(MessageEnum.SessionSaveMessage),
      getName(MessageEnum.SessionResponseMessage)
    ];

    this.state = {
      token: "",
      data: "",
      arrayHolder: []
    };
    // Customizable Area End
    runEngine.attachBuildingBlock(this as IBlock, this.subScribedMessages);
  }
 
  async receive(from: string, message: Message) {
    // Customizable Area Start
    switch (message.id) {
      case getName(MessageEnum.SessionResponseMessage):
        const token = message.getData(getName(MessageEnum.SessionResponseToken));
        this.setState({ token });
        this.getListRequest(token);
        break;

      case getName(MessageEnum.RestAPIResponceMessage):
        const apiRequestCallId = message.getData(getName(MessageEnum.RestAPIResponceDataMessage));
        if (apiRequestCallId !== this.getProductApiCallId) {
          break;
        }

        const responseJson = message.getData(getName(MessageEnum.RestAPIResponceSuccessMessage));
        if (responseJson && !responseJson.errors && responseJson.data) {
          const params = this.props.navigation.state.params;
          if (params?.data?.length > 0) {
            this.setState({ data: params.data });
          } else {
            this.setState({ data: responseJson.data });
          }
          this.setState({ arrayHolder: responseJson.data });
        } else {
          const errorReponse = message.getData(getName(MessageEnum.RestAPIResponceErrorMessage));
          this.showAlert("Error", errorReponse);
          this.parseApiCatchErrorResponse(errorReponse);
        }
        break;
    }
    // Customizable Area End
  }

  // Customizable Area Start
   async componentDidMount() {
    super.componentDidMount();
    this.getToken();
    if (this.isPlatformWeb() === false) {
      this.props.navigation.addListener("willFocus", () => {
        this.getToken();
      });
    }
  }

  getToken = () => {
    const msg: Message = new Message(
      getName(MessageEnum.SessionRequestMessage)
    );
    this.send(msg);
  };
  
  navigateToFilter = () => {
    let arrayHolder = this.state.arrayHolder;
    const priceRange = arrayHolder.map((element: any) => element.attributes.price);
    let min = Math.min.apply(null, priceRange);
    let max = Math.max.apply(null, priceRange);
    let params = this.props.navigation.state.params;
    if (params != undefined) {
      if (params.priceSelectedMin && params.priceSelectedMax) {
        this.props.navigation.push("Filteroptions", {
          min: min,
          max: max,
          priceSelectedMin: params.priceSelectedMin,
          priceSelectedMax: params.priceSelectedMax
        });
      }
    } else {
      this.props.navigation.push("Filteroptions", { min: min, max: max });
    }
  }

  getListRequest = (token: any) => {

    const header = {
      "Content-Type": configJSON.productApiContentType,
      token: token
    };
    const requestMessage = new Message(
      getName(MessageEnum.RestAPIRequestMessage)
    );
    this.getProductApiCallId = requestMessage.messageId;
    requestMessage.addData(
      getName(MessageEnum.RestAPIResponceEndPointMessage),
      configJSON.productAPiEndPoint
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestHeaderMessage),
      JSON.stringify(header)
    );
    requestMessage.addData(
      getName(MessageEnum.RestAPIRequestMethodMessage),
      configJSON.apiMethodTypeGet
    );
    runEngine.sendMessage(requestMessage.id, requestMessage);
  };
  // Customizable Area Start
  // Customizable Area End
}
// Customizable Area End
