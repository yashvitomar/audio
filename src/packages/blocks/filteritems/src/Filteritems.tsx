import React from "react";
// Customizable Area Start
import {
  StyleSheet,
  Text,
  Image,
  TouchableOpacity,
  View,
  ScrollView,
  TouchableWithoutFeedback,
  FlatList,
  SafeAreaView
} from "react-native";
// Customizable Area End

import FilteritemsController, { Props } from "./FilteritemsController";
import Icon from "react-native-vector-icons/FontAwesome";
export default class Filteritems extends FilteritemsController {
  constructor(props: Props) {
    super(props);
    // Customizable Area Start
    // Customizable Area End
  }

  // Customizable Area Start
  getList(item: any) {
    let value = item.item.attributes;
    console.log(value, "getList");
    return (
      <View style={styles.productBox}>
        <View style={styles.ImgContainer}>
          <Image
            style={styles.productImg}
            source={{ uri: value.images[0].url }}
            resizeMode="cover"
          />
        </View>
        <View style={styles.detailContent}>
          <Text style={styles.productName}>{value.name}</Text>
          <Text style={styles.price}>{value.price}</Text>
          <View style={[styles.flexBox, styles.starBox]}>
            <Text style={styles.rating}>{value.average_rating}</Text>
            <Icon
              name="star"
              size={12}
              style={styles.sortIcon}
              color="orange"
            />
          </View>
        </View>
      </View>
    );
  }
  // Customizable Area End

  render() {
    return (
      //Merge Engine DefaultContainer
      <SafeAreaView style={styles.container}>
      {/* Customizable Area Start */}
        <View style={styles.topContainer}>
          <TouchableOpacity
            style={[styles.topBox, styles.flexBox]}
            onPress={() => this.navigateToFilter()}
            testID={"navigateToFilter"}
          >
            <Icon
              name="filter"
              size={15}
              style={styles.sortIcon}
              color="#808080"
            />
            <Text style={styles.topText}>Filter</Text>
          </TouchableOpacity>
        </View>
        <ScrollView keyboardShouldPersistTaps="always">
          <TouchableWithoutFeedback
            onPress={() => {
              this.hideKeyboard();
            }}
          >
            {/* Customizable Area Start */}
            {/* Merge Engine UI Engine Code */}
            <View>
              <View style={[styles.productContainer, styles.flexBox]}>
                <FlatList
                  data={this.state.data}
                  renderItem={item => this.getList(item)}
                  extraData={this.state}
                  keyExtractor={(item: any) => item.id}
                  numColumns={2}
                  columnWrapperStyle={styles.row}
                />
              </View>
            </View>
            {/* Merge Engine UI Engine Code */}
            {/* Customizable Area End */}
          </TouchableWithoutFeedback>
        </ScrollView>
      {/* Customizable Area End */}
      </SafeAreaView>
      //Merge Engine End DefaultContainer
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginLeft: "auto",
    marginRight: "auto",
    width: "100%",
    maxWidth: 650,
    backgroundColor: "#ffffffff"
  },
  topContainer: {
    display: "flex",
    flexDirection: "row",
    borderBottomWidth: 1,
    borderBottomColor: "#ccc",
    shadowOffset: { width: 1, height: 1 },
    shadowColor: "black",
    shadowOpacity: 1.0,
    backgroundColor:"#FFF"
  },
  topBox: {
    width: "100%",
    paddingVertical: 5,
    marginVertical: 12,
    justifyContent: "center",
    alignItems: "center"
  },
  topText: {
    textAlign: "center",
    fontSize: 16,
    color: "#000"
  },
  rightBorder: {
    borderRightWidth: 1,
    borderRightColor: "#ccc"
  },
  modalContainer: {
    backgroundColor: "white",
    padding: 15,
    minHeight: 260,
    position: "relative"
  },
  modalBox: {
    margin: 0,
    width: "100%",
    justifyContent: "flex-end",
    marginBottom: -20
  },
  closeIcon: {
    position: "absolute",
    right: 15,
    top: 15,
    fontWeight: "normal",
    zIndex: 999
  },
  heading: {
    fontSize: 18,
    color: "#000",
    marginBottom: 15
  },
  sortList: {
    paddingVertical: 8,
    alignItems: "center"
  },
  row: {
    flex: 1,
    justifyContent: "space-between"
  },
  sortText: {
    fontSize: 16,
    color: "#696969"
  },
  sortIcon: {
    marginRight: 12
  },
  flexBox: {
    display: "flex",
    flexDirection: "row"
  },
  productContainer: {
    width: "100%",
    flex: 1,
    justifyContent: "center",
    backgroundColor: "white"
  },
  productBox: {
    height: 250,
    borderBottomWidth: 1,
    flexDirection: "column",
    flex: 0.5,
    borderRightWidth: 1,
    borderColor: "#ccc"
  },
  ImgContainer: {
    //marginBottom: 15,
    height: 150
  },
  productName: {
    paddingVertical: 5,
    fontSize: 16
  },
  price: {
    color: "#444",
    fontSize: 16,
    marginBottom: 10
  },
  productImg: {
    width: "100%",
    height: "100%"
  },
  detailContent: {
    padding: 10
  },
  rating: {
    color: "#000",
    paddingRight: 6
  },
  starBox: {
    alignItems: "center"
  }
});
// Customizable Area End
