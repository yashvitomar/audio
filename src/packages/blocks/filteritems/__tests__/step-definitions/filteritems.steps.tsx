import { defineFeature, loadFeature} from "jest-cucumber"
import { shallow, ShallowWrapper } from 'enzyme'

import * as helpers from '../../../../framework/src/Helpers'
import React from "react";
import Filteritems from "../../src/Filteritems"
const navigation = require("react-navigation")

const screenProps = {
    navigation: navigation,
    id: "Filteritems"
  }

const feature = loadFeature('./__tests__/features/filteritems-scenario.feature');

defineFeature(feature, (test) => {


    beforeEach(() => {
        jest.resetModules()
        jest.doMock('react-native', () => ({ Platform: { OS: 'web' }}))
        jest.spyOn(helpers, 'getOS').mockImplementation(() => 'web');
    });

    test('User navigates to filteritems', ({ given, when, then }) => {
        let filterBlocks:ShallowWrapper;
        let instance:Filteritems; 

        given('I am a User loading filteritems', () => {
            filterBlocks = shallow(<Filteritems {...screenProps}/>)
            expect(filterBlocks).toBeTruthy();
            instance = filterBlocks.instance() as Filteritems;
        });

        when('I navigate to the filteritems', () => {
             instance = filterBlocks.instance() as Filteritems
        });

        then('filteritems will load with out errors', () => {
            expect(filterBlocks).toBeTruthy()
            expect(filterBlocks).toMatchSnapshot()
        });

        then('I can leave the screen with out errors', () => {
            instance.componentWillUnmount()
            expect(filterBlocks).toBeTruthy()
            expect(filterBlocks).toMatchSnapshot()
        });
    });


});
