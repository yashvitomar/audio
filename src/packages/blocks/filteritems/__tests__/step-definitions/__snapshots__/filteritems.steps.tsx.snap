// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filteritems User navigates to filteritems 1`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 12,
            "paddingVertical": 5,
            "width": "100%",
          },
          Object {
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
      testID="navigateToFilter"
    >
      <Icon
        allowFontScaling={false}
        color="#808080"
        name="filter"
        size={15}
        style={
          Object {
            "marginRight": 12,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Filter
      </Text>
    </TouchableOpacity>
  </View>
  <ScrollViewMock
    keyboardShouldPersistTaps="always"
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <FlatList
            columnWrapperStyle={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
            data=""
            disableVirtualization={false}
            extraData={
              Object {
                "arrayHolder": Array [],
                "data": "",
                "token": "",
              }
            }
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={2}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </ScrollViewMock>
</SafeAreaView>
`;

exports[`filteritems User navigates to filteritems 2`] = `
<SafeAreaView
  style={
    Object {
      "backgroundColor": "#ffffffff",
      "flex": 1,
      "marginLeft": "auto",
      "marginRight": "auto",
      "maxWidth": 650,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#FFF",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "display": "flex",
        "flexDirection": "row",
        "shadowColor": "black",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 1,
      }
    }
  >
    <TouchableOpacity
      activeOpacity={0.2}
      onPress={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "marginVertical": 12,
            "paddingVertical": 5,
            "width": "100%",
          },
          Object {
            "display": "flex",
            "flexDirection": "row",
          },
        ]
      }
      testID="navigateToFilter"
    >
      <Icon
        allowFontScaling={false}
        color="#808080"
        name="filter"
        size={15}
        style={
          Object {
            "marginRight": 12,
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#000",
            "fontSize": 16,
            "textAlign": "center",
          }
        }
      >
        Filter
      </Text>
    </TouchableOpacity>
  </View>
  <ScrollViewMock
    keyboardShouldPersistTaps="always"
  >
    <TouchableWithoutFeedback
      onPress={[Function]}
    >
      <View>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "white",
                "flex": 1,
                "justifyContent": "center",
                "width": "100%",
              },
              Object {
                "display": "flex",
                "flexDirection": "row",
              },
            ]
          }
        >
          <FlatList
            columnWrapperStyle={
              Object {
                "flex": 1,
                "justifyContent": "space-between",
              }
            }
            data=""
            disableVirtualization={false}
            extraData={
              Object {
                "arrayHolder": Array [],
                "data": "",
                "token": "",
              }
            }
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={2}
            onEndReachedThreshold={2}
            removeClippedSubviews={false}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </ScrollViewMock>
</SafeAreaView>
`;
